### PR TITLE
Add `CODEOWNERS` file.

### PR DESCRIPTION
**Pull request by jonathanknowles**
## Issue

https://input-output.atlassian.net/browse/ADP-2785

## Summary

This PR adds a `CODEOWNERS` file, thus satisfying the requirement to identify the core maintainers of this repository.

### CODE REVIEW

**Review by Anviking — APPROVED** [review]

**Comment by Anviking** [comment]
(Approving with the concern that this may be noisy and drown specific review requests short-term)

**Comment by jonathanknowles** [comment]
@Anviking wrote:
> (Approving with the concern that this may be noisy and drown specific review requests short-term)

I also share this concern. Perhaps we can see how it goes, and then refine the `CODEOWNERS` file if it becomes too annoying?

In my experiments, it does seem possible, after creation of a PR, to reduce the size of the "reviewers" list to just those that you'd like to perform the review. I think GitHub only complains if the set of reviewers is completely disjoint from those listed in the `CODEOWNERS`.

**Pull request merged**
